### PR TITLE
(draft) backend-core typescript building successfully

### PR DESCRIPTION
**Pull request by Rory-Powell**
## Description
- backend-core typescript support
- convert some js files to ts, where the auto generated d.ts files were failing to build with server / worker
   -  packages/backend-core/src/db/Replication.ts
   - packages/backend-core/src/db/utils.ts
   - packages/backend-core/src/objectStore/index.ts

### CODE REVIEW

File: packages/backend-core/src/db/utils.ts [hunk 1]
```diff
@@ -1,47 +1,26 @@
...
   USER_BY_EMAIL: "by_email",
   BY_API_KEY: "by_api_key",
   USER_BY_BUILDERS: "by_builders",
 }
 
-exports.StaticDatabases = StaticDatabases
-
-exports.DocumentTypes = DocumentTypes
-exports.APP_PREFIX = APP_PREFIX
-exports.APP_DEV = exports.APP_DEV_PREFIX = APP_DEV
```
Comment (Rory-Powell) [inline packages/backend-core/src/db/utils.ts:38]: @mike12345567 not sure if I've converted this properly, I don't follow what the double `=` is doing. Below I've just re-exported conversions and contants for handiness. Probably we move `db/utils.ts` (and associated  Replication, constants, conversions) into -> `db/utils` and introduce `db/utils/index.ts` which will export things at the `utils` module level

Comment (mike12345567) [reply packages/backend-core/src/db/utils.ts:38]: The double equals is basically just equivalent to `a = b = 1` which means that `a = 1, b = 1`, essentially setting them all in one flow to the same value.

Comment (Rory-Powell) [reply packages/backend-core/src/db/utils.ts:38]: ah great, thanks!

**Comment by Rory-Powell** [comment]
closing this - finished on the backfill branch

**Pull request closed without merging**